### PR TITLE
fix(combobox): input value will now filter correctly as typed in

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #4348

## Summary

The Combobox dropdown does not properly filter the options as expected.

While filtering, the find function implements a lodash function `forIn` to iterate through passed in input properties. As is, it goes through all of the properties, including a property called guid. This is a longish letter string, and it’s matching almost everything typed in.

The solution here is to make a subset of the input object with only value and label properties before passing it in, so only relevant properties get to be iterated on.

### CODE REVIEW

File: src/components/combobox/combobox.e2e.ts [hunk 1]
```diff
@@ -62,28 +61,61 @@ describe("calcite-combobox", () => {
...
-  it.skip("should filter the items in listbox when typing into the input", async () => {
+  it("should filter the items in listbox when typing into the input", async () => {
     const page = await newE2EPage({
-      html: html` <calcite-combobox>
-        <calcite-combobox-item value="one" text-label="one"></calcite-combobox-item>
-        <calcite-combobox-item value="two" text-label="two"></calcite-combobox-item>
-      </calcite-combobox>`
+      html: html`
+        <calcite-combobox id="myCombobox">
+          <calcite-combobox-item value="Raising Arisona" text-label="Raising Arisona"></calcite-combobox-item>
```
Comment (jcfranco) [inline src/components/combobox/combobox.e2e.ts:68]: Nitpicks:

* Arizona
* Miller's

Sorry, can't help it. 😂

File: src/components/combobox/combobox.tsx [hunk 2]
```diff
@@ -677,7 +677,7 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
       item &&
       filteredData.some(({ label, value }) => {
         if (isGroup(item)) {
-          return value === item.label || value === item.label;
```
Comment (jcfranco) [inline src/components/combobox/combobox.tsx:680]: ✨🧹✨

**Comment by jcfranco** [comment]
Nice detective work, @Elijbet! 

I'm wondering if it's possible to get rid of `guid` instead. It got introduced [here](https://github.com/Esri/calcite-components/pull/1414/files), but doesn't seem to be used and it is automatically generated on the item. @paulcpederson Do you remember why [`guid`](https://github.com/Esri/calcite-components/blob/master/src/components/combobox/combobox.tsx#L818) is needed on the item data?

File: src/components/combobox/combobox.e2e.ts (hunk 1 above)
Comment (Elijbet) [reply src/components/combobox/combobox.e2e.ts:68]: I keep accumulating English grammar enthusiasts.

**Comment by paulcpederson** [comment]
@jcfranco let me look and see if I can remember

**Comment by paulcpederson** [comment]
@jcfranco @Elijbet Honestly I have no idea why that was added to the item data. As long as the active descendent aria information and the chip focusing work as before without it, I think it would be fine to remove.

**Review by jcfranco — APPROVED** [review]
:boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom:
:boom::bomb::bomb::bomb::bomb::boom::boom::bomb::bomb::boom::boom::boom::bomb::bomb::boom::boom::bomb::boom::boom::boom::bomb::boom::bomb::boom:
:boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::bomb::boom::bomb::bomb::boom::bomb::boom:
:boom::bomb::bomb::bomb::boom::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::bomb::boom::bomb::boom::bomb::boom:
:boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::bomb::boom::bomb::boom::boom::boom::bomb::boom::boom::boom:
:boom::bomb::bomb::bomb::bomb::boom::boom::bomb::bomb::boom::boom::boom::bomb::bomb::boom::boom::bomb::boom::boom::boom::bomb::boom::bomb::boom:
:boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom::boom:

**Review by benelan — CHANGES_REQUESTED** [review]
It looks like the fix was removed in [this commit](https://github.com/Esri/calcite-components/pull/4405/commits/77767537f6477fe6a2dc6b43e30d4f5bf2e11402)

**Review by benelan — APPROVED** [review]
Sorry, nevermind it looks like the fix changed.

**Comment by Elijbet** [comment]
> Sorry, nevermind it looks like the fix changed.

I tested it with screen reader also. It seems to work, but I don't have extensive experience with the screen reader. If it turns out something is off, I'll circle back to it.

**Pull request merged**